packages/client/machines/appMachine.ts

```diff
@@ -291,119 +291,198 @@ export function createAppMachine({
 
-                                            return event.data;
+                                        onError: {
+                                            target: 'debouncing',
                                         },
-                                    }),
+                                    },
+                                },
+
+                                debouncing: {
+                                    after: {
+                                        500: {
+                                            target: 'pollingMyProfileInformation',
+                                        },
+                                    },
                                 },
 
-                                onError: {
-                                    target: 'failedToRevalidateUserInformation',
+                                userEmailIsVerified: {
+                                    type: 'final',
                                 },
                             },
-                        },
 
-                        revalidatedUserInformation: {
-                            type: 'final',
+                            onDone: {
+                                target: '#app.reconnectingSocketConnection',
+                            },
                         },
 
-                        failedToRevalidateUserInformation: {},
-
                         signingOut: {
-                            invoke: {
-                                id: 'signOut',
+                            initial: 'idle',
+
+                            states: {
+                                idle: {},
 
-                                src: 'signOut',
+                                performSignOut: {
+                                    invoke: {
+                                        id: 'signOut',
 
-                                onDone: {
-                                    target: '#app.loadingAuthenticationTokenFromAsyncStorage',
-                                    actions:
-                                        'sendBroadcastReloadIntoBroadcastChannel',
+                                        src: 'signOut',
+
+                                        onDone: {
+                                            target: '#app.loadingAuthenticationTokenFromAsyncStorage',
+                                            actions:
+                                                'sendBroadcastReloadIntoBroadcastChannel',
+                                        },
+                                    },
+
+                                    on: {
+                                        SIGN_OUT: undefined,
+                                    },
                                 },
                             },
 
                             on: {
-                                SIGN_OUT: undefined,
+                                SIGN_OUT: {
+                                    target: '.performSignOut',
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Using an internal transition is definitely a good idea! I makes `SIGN_OUT: undefined` useless.